subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/http/BlockingHttpServer.java

```diff
@@ -98,6 +99,10 @@ public String callFromBuild(String resource) {
         return "System.out.println(\"calling " + uri + "\"); try { new java.net.URL(\"" + uri + "\").openConnection().getContentLength(); } catch(Exception e) { throw new RuntimeException(e); }; System.out.println(\"[G] response received\");";
     }
 
+    public String callFromTaskAction(String resource) {
```

Comment:
This looks a little confusing/tricky.  Is this a poor man's worker API kind of thing to release the locks?  When would a test want to do this?

Reply:
This is to avoid deadlocks.  The call to the blocking server creates synchronization.  If we release the project lock during configuration resolution (because we are stepping over to another project) and this task starts, it can block forever, preventing the task that released the project lock from being able to reacquire it (perhaps preventing the task it is waiting to synchronize with from starting).